server/actions_projects.go

```diff
@@ -110,21 +115,27 @@ func (s *RegistryServer) ListProjects(ctx context.Context, req *rpc.ListProjects
 		return nil, unavailableError(err)
 	}
 	defer s.releaseStorageClient(client)
+
+	if req.GetPageSize() < 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid page_size: must not be negative")
+	}
+
 	q := client.NewQuery(models.ProjectEntityName)
 	q, err = q.ApplyCursor(req.GetPageToken())
 	if err != nil {
-		return nil, internalError(err)
+		return nil, invalidArgumentError(err)
 	}
 	prg, err := createFilterOperator(req.GetFilter(),
 		[]filterArg{
+			{"name", filterArgTypeString},
 			{"project_id", filterArgTypeString},
 			{"display_name", filterArgTypeString},
 			{"description", filterArgTypeString},
 			{"create_time", filterArgTypeTimestamp},
 			{"update_time", filterArgTypeTimestamp},
 		})
 	if err != nil {
-		return nil, internalError(err)
```

Comment:
`createFilterOperator` already sets the status code for these errors, so I think it can be trusted right now. Moving forward it may be a good idea to only set response statuses in the RPC handlers, using other techniques for returning multiple types of errors internally.